crypto/cryptlib.c

```diff
@@ -319,14 +319,9 @@ void OPENSSL_die(const char *message, const char *file, int line)
  * pointers to volatile by preserving the reads and writes through them.
  * The latter is not required by the Standard if the memory pointed to
  * is not volatile.
- * Pointers themselves are volatile in the function signature to work
- * around a subtle bug in gcc 4.6+ which causes writes through
- * pointers to volatile to not be emitted in some rare,
- * never needed in real life, pieces of code.
  */
-int CRYPTO_memcmp(const volatile void * volatile in_a,
-                  const volatile void * volatile in_b,
-                  size_t len)
+# undef CRYPTO_memcmp
```

Comment:
BTW, why #undef? If you don't undef, then below implementations would be favoured by linker and it might be advantageous to use it instead of real memcmp. As mentioned earlier, if you compare two vectors one of which is oversized so that access to part of it will result in SEGV, real memcmp can fail to notice it, because first encountered difference can be in accessible part of vectors. But not below implementation. Well, this is provided that sanitizer doesn't actually deploy own memcmp that already does it...